bin/upgrade.py

```diff
@@ -188,6 +188,23 @@ def upgrade_statement(problem_path: Path, bar: ProgressBar) -> None:
             shutil.move(f, dest)
 
 
+def upgrade_output_validators(problem_path: Path, bar: ProgressBar) -> None:
+    if (problem_path / "output_validators").is_dir():
+        if (problem_path / "output_validator").exists():
+            bar.error(
+                "can't rename 'output_validators/', 'output_validator/' already exists", resume=True
+            )
+            return
+        content = [*(problem_path / "output_validators").iterdir()]
+        if len(content) == 1 and content[0].is_dir():
+            bar.log(f"renaming 'output_validators/{content[0].name}' to 'output_validator/'")
+            content[0].rename(problem_path / "output_validator")
```

Comment:
Note that if we rename the folder by moving it one level up, any symlink to `../../../../skel/validate.h` will fail to work. We may want to recreate this symlink by removing one `../`, using some magic with [`Path.readlink`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.readlink)?

Reply:
then we need to go through all files... i think its ok if the user has to fix this?

Reply:
I will see if copy is more useful then rename here ^^

Reply:
@mpsijm can you try the last commit?

Reply:
Yes, that works, thanks! :smile: